### DISCUSSION TITLE
GridLayout with equal-width columns

### DISCUSSION THREAD

**Discussion by dirkeinecke** [slint-ui/slint#6948] [Q&A] [answered]
I have the following elements:

```
        GroupBox {
            title: "Headline";
            width: 100%;
            GridLayout {
                spacing: 5px;
                Row {
                    Button {
                        text: "short";
                    }
                    Button {
                        text: "looooong";
                    }
                }
            }
        }
```

The two columns have different widths depending on the text width on the buttons. However, I want both columns to be of equal width. How can I achieve that?

Here is the example on [SlintPad](https://slintpad.com/?snippet=import+%7B+GroupBox%2C+Button+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++GroupBox+%7B%0A++++++++title%3A+%22Headline%22%3B%0A++++++++width%3A+100%25%3B%0A++++++++GridLayout+%7B%0A++++++++++++spacing%3A+5px%3B%0A++++++++++++Row+%7B%0A++++++++++++++++Button+%7B%0A++++++++++++++++++++text%3A+%22short%22%3B%0A++++++++++++++++%7D%0A++++++++++++++++Button+%7B%0A++++++++++++++++++++text%3A+%22looooong%22%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)

Dirk

**Comment by bjorn** [comment, marked as answer]
You can set both `min-width` and `preferred-with` to 0 (or some other equal values, if you want the text to fit) to make them equal as far as the layout is concerned: [SlintPad](https://slintpad.com/?snippet=import+%7B+GroupBox%2C+Button+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++GroupBox+%7B%0A++++++++title%3A+%22Headline%22%3B%0A++++++++width%3A+100%25%3B%0A++++++++GridLayout+%7B%0A++++++++++++spacing%3A+5px%3B%0A++++++++++++Row+%7B%0A++++++++++++++++Button+%7B%0A++++++++++++++++++++min-width%3A+0%3B%0A++++++++++++++++++++preferred-width%3A+0%3B%0A++++++++++++++++++++text%3A+%22short%22%3B%0A++++++++++++++++%7D%0A++++++++++++++++Button+%7B%0A++++++++++++++++++++min-width%3A+0%3B%0A++++++++++++++++++++preferred-width%3A+0%3B%0A++++++++++++++++++++text%3A+%22looooong%22%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native).
